cmd/ethrex/cli.rs

```diff
@@ -1,198 +1,276 @@
-                .default_value("jwt.hex")
-                .value_name("GATEWAY_JWTSECRET_PATH")
-                .action(ArgAction::Set),
-            )
-        } else {
-            cmd
+use crate::{
+    initializers::{init_blockchain, init_store},
+    utils::{self, set_datadir},
+    DEFAULT_DATADIR,
+};
+
+pub const VERSION_STRING: &str = env!("CARGO_PKG_VERSION");
+
+#[allow(clippy::upper_case_acronyms)]
+#[derive(ClapParser)]
+#[command(name="ethrex", author, version=VERSION_STRING, about, long_about = None)]
+pub struct CLI {
+    #[clap(flatten)]
+    pub opts: Options,
+    #[cfg(feature = "based")]
+    #[clap(flatten)]
+    pub based_opts: BasedOptions,
+    #[command(subcommand)]
+    pub command: Option<Subcommand>,
+}
+
+#[derive(ClapParser)]
+pub struct Options {
+    #[arg(
+        long = "http.addr",
+        default_value = "localhost",
+        value_name = "ADDRESS",
+        help_heading = "RPC options"
+    )]
+    pub http_addr: String,
+    #[arg(
+        long = "http.port",
+        default_value = "8545",
+        value_name = "PORT",
+        help_heading = "RPC options"
+    )]
+    pub http_port: String,
+    #[arg(long = "log.level", default_value_t = Level::INFO, value_name = "LOG_LEVEL", help_heading = "Node options")]
+    pub log_level: Level,
+    #[arg(
+        long = "authrpc.addr",
+        default_value = "localhost",
+        value_name = "ADDRESS",
+        help_heading = "RPC options"
+    )]
+    pub authrpc_addr: String,
+    #[arg(
+        long = "authrpc.port",
+        default_value = "8551",
+        value_name = "PORT",
+        help_heading = "RPC options"
+    )]
+    pub authrpc_port: String,
+    #[arg(
+        long = "authrpc.jwtsecret",
+        default_value = "jwt.hex",
+        value_name = "JWTSECRET_PATH",
+        help_heading = "RPC options"
+    )]
+    pub authrpc_jwtsecret: String,
+    #[arg(long = "p2p.enabled", default_value = if cfg!(feature = "l2") { "false" } else { "true" }, value_name = "P2P_ENABLED", action = ArgAction::SetTrue, help_heading = "P2P options")]
+    pub p2p_enabled: bool,
+    #[arg(
+        long = "p2p.addr",
+        default_value = "0.0.0.0",
+        value_name = "ADDRESS",
+        help_heading = "P2P options"
+    )]
+    pub p2p_addr: String,
+    #[arg(
+        long = "p2p.port",
+        default_value = "30303",
+        value_name = "PORT",
+        help_heading = "P2P options"
+    )]
+    pub p2p_port: String,
+    #[arg(
+        long = "discovery.addr",
+        default_value = "0.0.0.0",
+        value_name = "ADDRESS",
+        help_heading = "P2P options"
+    )]
+    pub discovery_addr: String,
+    #[arg(
+        long = "discovery.port",
+        default_value = "30303",
+        value_name = "PORT",
+        help_heading = "P2P options"
+    )]
+    pub discovery_port: String,
+    #[arg(
+        long = "network",
+        value_name = "GENESIS_FILE_PATH",
+        help_heading = "Node options"
+    )]
```

Comment:
Can we also mention that a known network name can be sent to use the preset genesis & boot nodes for it?.
We have this on the README as example:
Alternatively the name of a known network can be provided instead to use its preset genesis file and include its preset bootnodes. The networks currenlty supported include holesky, sepolia and mekong, (and I think we have ephemera too now).

Reply:
Of course!

Reply:
I'll do this for every flag in the README. This kind of help can be added as `long_help`.

Reply:
Done in [598f743](https://github.com/lambdaclass/ethrex/pull/2240/commits/598f743efdbf60c74b5b6b07458b933bf6464ffc). Let me know what you think.

Reply:
Also updated the README [5bc911d](https://github.com/lambdaclass/ethrex/pull/2240/commits/5bc911dd2bbf9cb69896509fcf076f2cb6cc81cb).

Reply:
Looks good! I think we should use lowercase for the network names as that is the format we recognize for the field

Reply:
Done [880f1ca](https://github.com/lambdaclass/ethrex/pull/2240/commits/880f1ca6d59e2f88438b25f096a3ba7efeab35bc)